test/buy-tshirt.spec.ts

```diff
@@ -1,41 +1,60 @@
+import { MenuContentPage, SignInPage, SummaryStepPage, AddresStepPage, ShippingStepPage,
+  PaymentStepPage, BankPaymentPage, OrderSummaryPage, ProductAddedPage, ProductListPage } from '../src/page';
 
 describe('Buy a t-shirt', () => {
+  const email: string = 'aperdomobo@gmail.com';
+  const password: string = 'WorkshopProtractor';
+
+  const menuContentPage: MenuContentPage = new MenuContentPage();
+  const summaryStepPage: SummaryStepPage = new SummaryStepPage();
+  const signInPage: SignInPage = new SignInPage(email, password);
+  const addresStepPage: AddresStepPage = new AddresStepPage();
+  const shippingStepPage: ShippingStepPage = new ShippingStepPage();
+  const paymentStepPage: PaymentStepPage = new PaymentStepPage();
+  const bankPaymentPage: BankPaymentPage = new BankPaymentPage();
+  const orderSummaryPage: OrderSummaryPage = new OrderSummaryPage();
+  const productAddedPage: ProductAddedPage = new ProductAddedPage();
+  const productListPage: ProductListPage = new ProductListPage();
+
   beforeEach(() => {
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
   });
 
   it('then should be bought a t-shirt', async () => {
     await browser.get('http://automationpractice.com/');
-    await(browser.sleep(10000));
-    await $('#block_top_menu > ul > li:nth-child(3) > a').click();
+    await(browser.sleep(5000));
+
+    await menuContentPage.goToTShirtMenu();
     await(browser.sleep(3000));
-    await $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default').click();
+
+    await productAddedPage.addProduct();
     await(browser.sleep(3000));
-    await $('[style*="display: block;"] .button-container > a').click();
+
+    await productListPage.proceedToCheckOut();
     await(browser.sleep(3000));
-    await $('.cart_navigation span').click();
+
+    await summaryStepPage.completeSummaryStep();
     await(browser.sleep(3000));
 
-    await $('#email').sendKeys('aperdomobo@gmail.com');
-    await $('#passwd').sendKeys('WorkshopProtractor');
-    await $('#SubmitLogin > span').click();
+    await signInPage.completeSignInStep();
     await(browser.sleep(3000));
 
-    await $('#center_column > form > p > button > span').click();
+    await addresStepPage.completeAddresStepPage();
     await(browser.sleep(3000));
 
-    await $('#cgv').click();
+    await shippingStepPage.agreeTermsShippingStep();
     await(browser.sleep(3000));
 
-    await $('#form > p > button > span').click();
+    await shippingStepPage.completeShippingStep();
     await(browser.sleep(3000));
-    await $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a').click();
+
+    await paymentStepPage.goToPayByBank();
     await(browser.sleep(3000));
-    await $('#cart_navigation > button > span').click();
+
+    await bankPaymentPage.confirmOrder();
     await(browser.sleep(3000));
 
-    await expect($('#center_column > div > p > strong').getText())
-      .toBe('Your order on My Store is complete.');
+    orderSummaryPage.confirmIfTheOrderWasSuccesful();
```

Comment:
The idea of the POM is to model the business behavior of an app component, however, the expectations or validations should remain in the test classes. That being said, the right pattern here should be to have a method to retrieve the orderConfirmation from the summary and compare that confirmation with the expected message. I will add this review to your current PR.